pkg/registry/service/rest.go

```diff
@@ -265,6 +276,97 @@ func (*REST) NewList() runtime.Object {
 	return &api.ServiceList{}
 }
 
+func (rs *REST) healthCheckNodePortUpdate(oldService, service *api.Service) (bool, error) {
+	// Health Check Node Port handling during updates
+	//
+	// Case 1. Transition from globalTraffic to OnlyLocal for the ESIPP annotation
+	//
+	//   Allocate a health check node port or attempt to reserve the user-specified one, if provided.
+	//   Insert health check node port as an annotation into the service's annotations
+	//
+	// Case 2. Transition from OnlyLocal to Global for the ESIPP annotation
+	//
+	//   Free the existing healthCheckNodePort and clear the health check nodePort annotation
+	//
+	// Case 3. No change (Global ---stays--> Global) but prevent invalid annotation manipulations
+	//
+	//   Reject insertion of the "service.alpha.kubernetes.io/healthcheck-nodeport" annotation
+	//
+	// Case 4. No change (OnlyLocal ---stays--> OnlyLocal) but prevent invalid annotation manipulations
+	//
+	//   Reject deletion of the "service.alpha.kubernetes.io/healthcheck-nodeport" annotation
+	//   Reject changing the value of the healthCheckNodePort annotation
+	//
+	oldServiceHasHealthCheckNodePort := shouldCheckOrAssignHealthCheckNodePort(oldService)
+	oldHealthCheckNodePort := apiservice.GetServiceHealthCheckNodePort(oldService)
+
+	assignHealthCheckNodePort := shouldCheckOrAssignHealthCheckNodePort(service)
```

Comment:
the name of this function is confusing and makes me think it's doing 2 different things. Can you clarify the name or add a comment above it. It returns a bool, so the shouldCheck part doesn't make sense to me.